src/main/java/runhistoryplus/patches/PotionUseAndDiscardRunHistoryPatch.java

```diff
@@ -179,4 +183,33 @@ public int[] Locate(CtBehavior ctMethodToPatch) throws CannotCompileException, P
             }
         }
     }
+
+    @SpirePatch(clz = ObtainPotionAction.class, method = "update")
```

Comment:
It looks to me like this approach will still miss potions obtained from Entropic Brew when its used outside combat, since that goes through ObtainPotionEffect (instead of ObtainPotionAction). The code between the two is very similar. Not sure if you tested that case (or even intended to handle it).

Looks like Entropic Brew is the only thing that uses ObtainPotionEffect, so adding logs there would be straightforward.

The fact that Entropic Brew can be used in or out of combat means that it might make sense to log Entropic Brew and Alchemize separately (and there are some other reasons to consider that, which I'll put in a different comment). If we did that and we were paranoid about other mods that may use ObtainPotionEffect, we could pass through something indicating that it came from Entropic Brew.

Reply:
I totally missed ObtainPotionEffect, thanks for the hint!